geoserver/geoserver_seeder/geoserver_seeder.rb

```diff
@@ -357,87 +49,74 @@ def seed_layers(layers, opts)
 #   - true on success, false otherwise
 #
 def geoserver_seeder_main(opts)
-  layers = []
 
-  if opts[:geonetwork]
-    # Get layers from Geonetwork URL
-    geonetwork_connector = GeonetworkConnector.new(opts[:geonetwork], opts[:geoserver])
-    layers = geonetwork_connector.wms_layers
-  else
-    # Get layers from command line
-    layers = opts[:layers]
-  end
+    geonetwork_connector = GeonetworkConnector.new(opts)
+
+    # layers that match supplied WMS server or all layers on Geonetwork
+    geonetwork_layers = geonetwork_connector.wms_layers
+    layers_to_seed = []
+
+    # add only layers from command line
+    if opts[:layers]
+        opts[:layers].each do | layer|
+            if(geonetwork_layers.any? {|geolayer| geolayer[1] == layer})
+                layers_to_seed.push([opts[:geoserver], layer])
```

Comment:
The geoserver address was used as a predicate when matching in the geonetwork record - in order not to seed all geoserver layers. It would be good to get @pmbohm's input on how these scripts are expected to be called.